lib/routes/devices-and-sessions.js

```diff
@@ -237,6 +243,95 @@ module.exports = (log, db, config, customs, push, devices) => {
         }
       }
     },
+    {
+      method: 'GET',
+      path: '/account/device/messages',
+      config: {
+        validate: {
+          query: {
+            service: isA.string().valid('sendtab').required(),
+            index: isA.string().optional(),
+            limit: isA.number().optional(),
+          }
+        },
+        auth: {
+          strategy: 'sessionToken'
+        }
+      },
+      handler (request, reply) {
+        log.begin('Account.deviceMessages', request)
+
+        const sessionToken = request.auth.credentials
+        const uid = sessionToken.uid
+        const deviceId = sessionToken.deviceId
+        const ip = request.app.clientAddress
+        const {service, index, limit} = request.query
+
+        return customs.checkAuthenticated('/device/messages', ip, uid)
+          .then(() => oAuthMint.getOAuthToken(sessionToken, PUSHBOX_OAUTH_SCOPE_RECV))
+          .then(resp => {
+            const {access_token: oAuthToken} = resp
+            return pushbox.retrieve(uid, deviceId, service, {index, limit}, oAuthToken)
+          })
+          .then(reply, reply)
+      }
+    },
+    {
+      method: 'POST',
+      path: '/account/device/messages',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            to: isA.string().required(),
+            service: isA.string().valid('sendtab').required(),
+            data: isA.string().regex(validators.URL_SAFE_BASE_64).required()
+          }
+        },
+        response: {
+          schema: {}
+        }
+      },
+      handler (request, reply) {
+        log.begin('Account.deviceMessages', request)
+
+        const body = request.payload
+        const sessionToken = request.auth.credentials
+        const uid = sessionToken.uid
+        const ip = request.app.clientAddress
+        const {to, data, service} = body
+
+        return customs.checkAuthenticated('/device/messages', ip, uid)
```

Comment:
👍 to rate-limiting this; do we need a customs-server patch to add support for the new action type?

Also, I think the first argument here (the "action") is usually named like "deviceMessages" rather than "/device/message".  I'd use something like "sendDeviceMessage" in this case.